tensorflow_datasets/structured/heart_disease.py

```diff
@@ -0,0 +1,176 @@
+  return int(float(d))
+
+def convert_to_float(d):
+  return float(d)
+
+def return_same(d):
+  return d
+
+FEATURE_DICT = collections.OrderedDict([
+    ("age", (tf.int32, convert_to_int)),
+    ("sex", (tfds.features.ClassLabel(names=['female', 'male']),
+             lambda d: convert_to_label(d, _SEX_DICT))),
+    ("cp", (tfds.features.ClassLabel(names=_CP_NAMES),
+            lambda d: convert_to_label(d, _CP_DICT))),
+    ("trestbps", (tf.int32, convert_to_int)),
+    ("chol", (tf.int32, convert_to_int)),
+    ("fbs", (tfds.features.ClassLabel(names=['false', 'true']),
+             lambda d: convert_to_label(d, _FBS_DICT))),
+    ("restecg", (tf.int32, convert_to_int)),
+    ("thalach", (tf.int32, convert_to_int)),
+    ("exang", (tfds.features.ClassLabel(names=['no', 'yes']),
+               lambda d: convert_to_label(d, _EXE_DICT))),
+    ("oldpeak", (tf.float32, convert_to_float)),
+    ("slope", (tfds.features.ClassLabel(names=_SLOPE_NAMES),
+               lambda d: convert_to_label(d, _SLOPE_DICT))),
+    ("ca", (tf.int32, convert_to_int)),
+    ("thal", (tfds.features.ClassLabel(names=_THAL_NAMES),
+              lambda d: convert_to_label(d, _THAL_DICT)))
+])
+
+
+class HeartDisease(tfds.core.GeneratorBasedBuilder):
+  """Heart disease dataset with 13 attributes."""
+
+  VERSION = tfds.core.Version("0.0.1", "New split API (https://tensorflow.org/datasets/splits)")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "features": {name: dtype
+                         for name, (dtype, func) in FEATURE_DICT.items()},
+            "label": tfds.features.ClassLabel(names=['0', '1', '2', '3', '4'])
+        }),
+        supervised_keys=("features", "label"),
+        homepage='http://archive.ics.uci.edu/ml/datasets/Heart+Disease',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+
+    filepath = dl_manager.download(_DOWNLOAD_URL)
+
+    # There is no predefined train/val/test split for this dataset.
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={"filepath": filepath}
+            ),
+        ]
+
+  def _generate_examples(self, filepath):
+    """Yields examples."""
+
+    feature_columns = ["age", "sex", "cp", "trestbps", "chol", "fbs", "restecg",
+                       "thalach", "exang", "oldpeak", "slope", "ca", "thal"]
+
+    with tf.io.gfile.GFile(filepath) as f:
+      all_lines = f.read().splitlines()
+      records = [l for l in all_lines if ('?' not in l) and l]
+      for i, row in enumerate(records):
+        features = row.split(',')
+        label = features.pop()
+        yield i, {
+            "features": {
+                feature_columns[col]:\
+                FEATURE_DICT[feature_columns[col]][1](value)
+                for col, value in enumerate(features)
```

Comment:
for name, value in zip(feature_columns, features)

Reply:
thanks,